### ISSUE TITLE
Ability to use OCI Helm Chart Registry in Apps&Marketplace

### ISSUE THREAD

**Issue by rohitsakala** [rancher/dashboard#9815]
Internal Reference: SURE-7573

**Is your feature request related to a problem? Please describe.**
Helm supports OCI-based helm charts since 3.8.0 and Rancher doesn't through Apps & Markatplace. Backend is being done for it through this RFC https://confluence.suse.com/display/RMI/Support+OCI+Registry+based+Helm+Charts

**Describe the solution you'd like**
RFC https://confluence.suse.com/display/RMI/Support+OCI+Registry+based+Helm+Charts

**UI Specific**
- [x] https://confluence.suse.com/display/RMI/Support+OCI+Registry+based+Helm+Charts
- [x] The add cluster repo page should contain a new target named `oci url` in addition to http and git and only show basic auth as oci only supports basic auth right now.
- [x] Add experimental banner for OCI as this feature is experimental
- [x] There is a way to not show Created date of index in the Individual Chart page if the Created date is [zero](https://golang.cafe/blog/golang-check-if-time-is-empty-example.html#:~:text=Golang%20Zero%20Values,00%3A00%20%2B0000%20UTC%20.)
- [x] There is a way to specify [plainHTTP](https://github.com/rancher/rancher/blob/release/v2.9/pkg/apis/catalog.cattle.io/v1/types.go#L31) checkbox if OCI target is selected.
- [ ] There is a way to specify that users are recommended to add only OCI URLs that contain *only* helm charts. For example oci://dp.apps.rancher.io/charts or oci://registry-1.docker.io/rohitsakala/testingchart. 

`OCI URL's must contain ONLY helm charts. For example oci://test.rancher.io/charts or oci://test.rancher.io/charts/mychart. The list of charts available from the repository are updated every 6 hours.`

- [x] In the individual Charts page, the UI must fetch the `Application Version`, `Home`, `Maintainers` values from the response of the `link=info` API call.
- [ ] There is a way to specify the ExponentialBackOff values in the Create ClusterRepo page from the user to fill in this struct 
```
type ExponentialBackOffValues struct {
	MinWait    int  `json:"minWait,omitempty"`
	MaxWait    int `json:"maxWait,omitempty"`
	MaxRetries int     `json:"maxRetries,omitempty"`
}
``` 
  which is in the spec of the clusterRepo. These are optional values and the default values set in the backend from 1s,5s,5 
  respectively. The minimum is 1 second 
- [x] When the individual chart page is loaded, if the backend sends an error, that error should be shown in the UI.
- [x] There is a way to specify [caBundle](https://github.com/rancher/rancher/blob/release/v2.9/pkg/apis/catalog.cattle.io/v1/types.go#L52) and [InsecureSkipTLSverify](https://github.com/rancher/rancher/blob/release/v2.9/pkg/apis/catalog.cattle.io/v1/types.go#L55) for all targets.

#### These tasks will be delayed since docs page needs to be setup first.

- [ ] There is a way to show docs page for OCI 
- [ ] There is way that ExponentailBackOffValues struct will show have a docs link pointing on how to use it and how it works.

**Comment by gunamata** [comment]
JIRA-3173

**Comment by richard-cox** [comment]
@rohitsakala is there a target milestone for the feature?

**Comment by rohitsakala** [comment]
@richard-cox Yes it would be Q1-2.8 Rancher release

**Comment by gaktive** [comment]
Waiting on the backend to be ready before working on this. Checking with @gunamata on what tickets they have to track on their side.

**Comment by gunamata** [comment]
@gaktive , Here's the backend ticket - https://github.com/rancher/rancher/issues/29105

**Comment by CC007** [comment]
I assume that this is no longer targeted for 2.8.0?

**Comment by rohitsakala** [comment]
@richard-cox Yes the values can be 

1. oci://test.acure.io/charts/etcd:1.0.0
2. oci://test.acure.io/charts/etcd
3. oci://test.acure.io/charts
4. oci://test.acure.io

@mattfarina suggested to use only one URL field for specifying OCI URL since fleet also gives a URL field. This maintains consistency.

**Comment by momesgin** [comment]
@kwwii based on the requirements, I have some ux/ui questions:

- Should we go with separate inputs for different parts of the url: `(registry_host_url) / (repository_namespace) : (tag)`
- How should we display these information to the user:

    > Add experimental banner for OCI as this feature is experimental

    > There is a way to specify that users are recommended to add only OCI URLs that contain only helm charts for better performance...

- Also there are three more values that need to be specified for `ExponentialBackOff`: `MinWait`, `MaxWait` and `MaxRetries`. A UI mock might be helpful for the whole section.

**Comment by momesgin** [comment]
@rohitsakala should UI do something about this item?
> There is a way to not show Created date of index in the Individual Chart page if the Created date is [zero]

**Comment by richard-cox** [comment]
@momesgin RE `Should we go with separate inputs for different parts of the url: (registry_host_url) / (repository_namespace) : (tag)` i think in planning the vote was to keep this as a single field

**Comment by kwwii** [comment]
@momesgin Be sure to add a proper placeholder to reflect the url/namespace:tag information the user needs to enter

**Comment by rohitsakala** [comment]
@momesgin Yes, index.yaml is recieved by the UI and the UI should not display the date&time in the `List of Versions` column when the time field is zero or empty.
